Source/WebKit/Platform/cocoa/SharedMemoryCocoa.cpp

```diff
@@ -130,28 +114,23 @@ void SharedMemory::Handle::setOwnershipOfMemory(const WebCore::ProcessIdentity&
 
 bool SharedMemory::Handle::isNull() const
 {
-    return !m_port;
+    return !m_handle;
 }
 
 void SharedMemory::Handle::clear()
 {
-    if (m_port)
-        deallocateSendRightSafely(m_port);
-
-    m_port = MACH_PORT_NULL;
-    m_size = 0;
+    *this = { };
 }
 
 void SharedMemory::Handle::encode(IPC::Encoder& encoder) const
 {
     encoder << static_cast<uint64_t>(m_size);
-    encoder << MachSendRight::adopt(m_port);
-    m_port = MACH_PORT_NULL;
+    encoder << WTFMove(m_handle); // FIXME: add rvalue encode.
```

Comment:
Is WTFMove doing anything here, or it just invoking the copy operator of encoder? This is const function, so we should be copying the send right into the encoder.

Seems like it'd be clearer to just not have WTFMove.

Reply:
We should be copying, but as the added copy and the previous code mentions, the intention is to move the mutable m_handle into the encoder, similar what was before.